pkg/cgroups/cgroups.go

```diff
@@ -178,6 +203,103 @@ func (subsys Subsystems) AddPID(name string, pid int) error {
 	return appendIntToFile(filepath.Join(subsys.CPU, name, "cgroup.procs"), pid)
 }
 
+func CreatePodCgroup(cgroupName string, config Config) error {
+	subsys, err := Find()
+	if err != nil {
+		return err
+	}
+	node, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+	err = os.MkdirAll(filepath.Join(subsys.CPU, "p2", node), 0755)
+	if err != nil && !os.IsExist(err) {
+		return err
+	}
+	err = os.MkdirAll(filepath.Join(subsys.Memory, "p2", node), 0755)
+	if err != nil && !os.IsExist(err) {
+		return err
+	}
+	return create(cgroupName, "", config)
+}
+
+func CreateLaunchableCgroup(cgroupName, podID string, config Config) error {
+	return create(cgroupName, podID, config)
+}
+
+func create(cgroupName string, podID string, config Config) error {
+	config.Name = cgroupName
+
+	var subsys Subsystems
+	var err error
+	if podID != "" {
+		subsys, err = FindWithParentGroup(podID)
+	} else {
+		subsys, err = Find()
+	}
+
+	if err != nil {
+		return util.Errorf("Could not find cgroupfs mount point: %s", err)
+	}
+
+	err = subsys.Write(config)
+	if _, ok := err.(UnsupportedError); ok {
+		// if a subsystem is not supported, just log
+		// and carry on
+		fmt.Printf("Unsupported subsystem (%s), continuing\n", err)
```

Comment:
now that this isn't in a script anymore, it doesn't really make sense here. Maybe return a special error type all the way up and check it in `p2-exec`